### ISSUE TITLE
Replacing text changes file encoding

### ISSUE THREAD

**Issue by stovocor** [rtyley/bfg-repo-cleaner#251]
We are in the process of shrinking a large repository while also replacing CRLF line feeds. For this, we use the text replacement option:

`$ java -jar bfg-1.12.16.jar --no-blob-protection --replace-text replacements.txt repo`

The replacements file contains a single line:

`regex:\r(\n)==>$1`

However, this has a side effect that the encoding of some files is changed. For example, a file in the original repository is reported as

`ISO-8859 text, with CRLF line terminators`

by the Unix file command, while the "cleaned" repository contains the file as

`UTF-8 Unicode text`

A hexdump confirms that the encoding has indeed changed. Providing the option `-Dfile.encoding=ISO-8859-1` does not change anything.

**Comment by stovocor** [comment]
Okay, I found the cause of this issue. `QuickBlobCharsetDetector` uses the first 8.000 bytes which is not enough for some of our classes. If the first 8.000 bytes are ASCII but somewhere in the rest of the file a character in ISO-8859-1 is present, the detector will return UTF-8.

I "fixed" this by changing the charset order in the detector, but that is definitely not useful for other repositories.

BTW, `sbt assembly` reports failed tests on the unchanged master, is this normal?

**Comment by javabrett** [comment]
> BTW, sbt assembly reports failed tests on the unchanged master, is this normal?
Nope, `master` build is passing for me [and on Travis](https://travis-ci.org/rtyley/bfg-repo-cleaner/builds/289392706).

Which commit are you building, using which command and what is the test-failure output?

**Comment by stovocor** [comment]
> Which commit are you building,

Commit 94225fdc6dfea01bc1de7517f5edc4c7ac81d7fd (Tag v1.12.16)

> using which command

% sbt assembly

> and what is the test-failure output?

[info] Size formatter
[info] - should correctly format *** FAILED ***
[info]   "1[,]0 KB" was not equal to "1[.]0 KB" (ByteSizeSpecs.scala:55)

I attached the complete output.

[sbt-assembly-94225fd.txt](https://github.com/rtyley/bfg-repo-cleaner/files/1551520/sbt-assembly-94225fd.txt)

**Comment by stovocor** [comment]
That works as expected. However, in my opinion the test case should be agnostic to the locale environment.
